### PR TITLE
feat: Adds new purchase order fields to invoices

### PR DESCRIPTION
**Pull request by warmwaffles**
- Added `lineNumber` to `#/components/schemas/CreatePurchaseOrderItem`
- Added `lineNumber` to `#/components/schemas/PurchaseOrderItem`
- Renamed `quantity` to `quantityRequested` in `#/components/schemas/CreatePurchaseOrderItem`
- Renamed `quantity` to `quantityRequested` in `#/components/schemas/PurchaseOrderItem`
- Added `quantityReceived` to `#/components/schemas/CreatePurchaseOrderItem`
- Added `quantityReceived` to `#/components/schemas/PurchaseOrderItem`
- Added `quantityInvoiced` to `#/components/schemas/InvoiceLineItem`
- Added `quantityInvoiced` to `#/components/schemas/CreateInvoiceLineItem`
- Added `quantityInvoiced` to `#/components/schemas/TrainingInvoiceLineItemUpsert`
- Added `lineItemTotal` to `#/components/schemas/InvoiceLineItem`
- Added `poLineNumber` to `#/components/schemas/InvoiceLineItem`
- Added `poNumber` to `#/components/schemas/InvoiceLineItem`

### CODE REVIEW

File: vic.api.v0.yaml [hunk 1]
```diff
@@ -2193,7 +2198,7 @@ components:
           nullable: true
           example: "kg"
           maxLength: 255
-        quantity:
+        quantityReceived:
           type: string
           format: decimal
           example: "12.3"
```
Comment (GrantZukowski) [inline vic.api.v0.yaml:2204]: curious if there is a limit to how long these are if they are format decimal.

**Review by GrantZukowski — APPROVED** [review]

**Comment by GrantZukowski** [comment]
:shipit:

File: vic.api.v0.yaml (hunk 1 above)
Comment (warmwaffles) [reply vic.api.v0.yaml:2204]: As it stands, we will store the full precision and return the full precision.

**Review by jonathanpglick — APPROVED** [review]
Looks right to me!

**Pull request merged**